renderer/src/pages/Onboarding.tsx

```diff
@@ -20,20 +20,24 @@ const OnboardingPage = (): JSX.Element => {
   const navigate = useNavigate()
   const [isLoading, setIsLoading] = useState<boolean>(true)
   const [isOnboardingCompleted, setIsOnboardingCompleted] = useState<boolean|null>()
+  const [filAddress, setFilAddress] = useState<string|null>()
 
   useEffect(() => {
     (async () => {
       await sleep(2000)
+      setFilAddress(await getFilAddress())
       setIsOnboardingCompleted(await getOnboardingCompleted())
       setIsLoading(false)
     })()
   }, [])
 
   useEffect(() => {
-    if (isOnboardingCompleted) {
+    if (isOnboardingCompleted && !filAddress) {
       navigate('/wallet', { replace: true })
+    } else if (filAddress) {
+      navigate('/dashboard', { replace: true })
     }
-  }, [isOnboardingCompleted, navigate])
+  }, [filAddress, isOnboardingCompleted, navigate])
```

Comment:
This is a suggestion for lines 23 - 40 (of the new file):

```js
  useEffect(() => {
    (async () => {
      await sleep(2000)
      if (await getFilAddress()) {
        return navigate('/dashboard', { replace: true })
      }
      setIsOnboardingCompleted(await getOnboardingCompleted())
      setIsLoading(false)
    })()
  }, [navigate])

  useEffect(() => {
    if (isOnboardingCompleted) {
      navigate('/wallet', { replace: true })
    } 
  }, [isOnboardingCompleted, navigate])
````

I believe there's no advantage in having `filAddress` as a state value